pkg/virtctl/vmexport/vmexport_test.go

```diff
@@ -0,0 +1,557 @@
+			err := cmd()
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).Should(Equal("Processing failed: Test error"))
+		})
+
+		It("VirtualMachineExport download fails when there's no volume available", func() {
+			testInit(http.StatusOK)
+			vme := vmexportSpec(vmexportName, defaultNamespace, "pvc", "test-pvc")
+			vme.Status = getVMELinks(nil)
+			expectVMExportGet(vmExportClient, vme)
+
+			cmd := clientcmd.NewRepeatableVirtctlCommand(commandName, DOWNLOAD, vmexportName, setflag(OUTPUT_FLAG, "disk.img"), setflag(VOLUME_FLAG, volumeName))
+			err := cmd()
+			Expect(err).To(HaveOccurred())
+			expectedError := fmt.Sprintf("Unable to access the volume info from '%s/%s' VirtualMachineExport", defaultNamespace, vmexportName)
+			Expect(err.Error()).Should(ContainSubstring(expectedError))
+		})
+
+		It("VirtualMachineExport download fails when the volume has a different name than expected", func() {
+			testInit(http.StatusOK)
+			vme := vmexportSpec(vmexportName, defaultNamespace, "pvc", "test-pvc")
+			vme.Status = getVMELinks([]exportv1.VirtualMachineExportVolume{
+				{
+					Name: "no-test-volume",
+					Formats: []exportv1.VirtualMachineExportVolumeFormat{
+						{
+							Format: exportv1.KubeVirtRaw,
+							Url:    server.URL,
+						},
+					},
+				},
+			})
+			expectVMExportGet(vmExportClient, vme)
+
+			cmd := clientcmd.NewRepeatableVirtctlCommand(commandName, DOWNLOAD, vmexportName, setflag(OUTPUT_FLAG, "disk.img"), setflag(VOLUME_FLAG, volumeName))
+			err := cmd()
+			Expect(err).To(HaveOccurred())
+			expectedError := fmt.Sprintf("Unable to get a valid URL from '%s/%s' VirtualMachineExport", defaultNamespace, vmexportName)
+			Expect(err.Error()).Should(ContainSubstring(expectedError))
+		})
+
+		It("VirtualMachineExport download fails when no format is available", func() {
+			testInit(http.StatusOK)
+			vme := vmexportSpec(vmexportName, defaultNamespace, "pvc", "test-pvc")
+			vme.Status = getVMELinks([]exportv1.VirtualMachineExportVolume{{Name: volumeName}})
+			expectVMExportGet(vmExportClient, vme)
+
+			cmd := clientcmd.NewRepeatableVirtctlCommand(commandName, DOWNLOAD, vmexportName, setflag(OUTPUT_FLAG, "disk.img"), setflag(VOLUME_FLAG, volumeName))
+			err := cmd()
+			Expect(err).To(HaveOccurred())
+			expectedError := fmt.Sprintf("Unable to get a valid URL from '%s/%s' VirtualMachineExport", defaultNamespace, vmexportName)
+			Expect(err.Error()).Should(ContainSubstring(expectedError))
+		})
+
+		It("VirtualMachineExport download fails when the only available format is incompatible with download", func() {
+			testInit(http.StatusOK)
+			vme := vmexportSpec(vmexportName, defaultNamespace, "pvc", "test-pvc")
+			vme.Status = getVMELinks([]exportv1.VirtualMachineExportVolume{
+				{
+					Name: volumeName,
+					Formats: []exportv1.VirtualMachineExportVolumeFormat{
```

Comment:
A thought here, I see this piece repeated a bunch of times:
```go
[]exportv1.VirtualMachineExportVolumeFormat{
  {
    Format: exportv1.Dir,
    Url:    server.URL,
  },
},
```
For the different formats. Might make sense to have these as a function, or const if server.URL is contant. And you can just put in the function/const instead of repeating those 4 lines over and over.